server-phoenix/lib/helios/widget.ex

```diff
@@ -35,4 +39,9 @@ defmodule Helios.Widget do
     tod = DateTime.to_time(time)
     from q in query, where: (q.start <= ^tod or is_nil(q.start)) and (q.stop >= ^tod or is_nil(q.stop))
   end
+
+  def next_or_default(query, current_id) do
+    queryable = from q in query, where: (q.id > ^current_id)
```

Comment:
This contains the same bug from ruby. We can't assume the sort is based on id. We should fetch by id and then get the position to define our greater than where clause.